---
title: "B viburni genome based -  GENE LEVEL DIFFERENTIAL EXPRESSION ANALYSIS"
author: "Isabelle Vea"
date: "21/10/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r}
#packages
#if (!requireNamespace("BiocManager", quietly = TRUE))
 #   install.packages("BiocManager")
library(edgeR)
library(methods)
library(dplyr)
library(tidyverse)
library(limma)

#BiocManager::install("Glimma")
library(Glimma)
library(gplots)
#BiocManager::install("org.Mm.eg.db")
library(org.Mm.eg.db)
```

# Step 1: Annotation file
```{r}
genetotranscript<-read.delim("Downloads/p.viburni.freeze.v0.braker.transcripts.to.genes.txt", header=FALSE)
genetotranscript <-genetotranscript[order(genetotranscript$V1),]

interpro<-read.delim("Downloads/p.viburni.freeze.v0.braker.interproscan.tsv", header = FALSE) #GO terms with transcript id but no gene id
interpro <-interpro[order(interpro$V1),]

sampleinfo<-read.csv("/Users/isabelle/Dropbox/_Github_reps/B_viburni/R_scripts/sampleinfoPviburniB.csv") #sample group info

head(genetotranscript,2) #col1 is gene id, col 2 is transcript id
names(genetotranscript)<-c("gid","tid") #gid=gene id, tig = transcript id

head(interpro,2) #column V1 is transcript id
names(interpro)[names(interpro) == 'V1'] <- 'tid'
head(interpro,2)

annot = merge(genetotranscript,interpro,by.x="tid", by.y="tid",all=TRUE)
head(annot)
```


# Step 2: Count file from all samples
```{r}
#need a dataframe containing all gene info per gene id
a<-read.delim("Downloads/RSEM_digi.counts.matrix",header=TRUE) #matrix generated from rsem
head(a,2) 

colnames(a) <- substr(colnames(a),start=1,stop=6) #removing ".genes.results" in colnames
colnames(a)

#first column X is the gene id, this needs to be removed but has the order of gene id has to match when merging with gene info

a1 <- a[,-1]
head(a1)

rownames(a1) <- a[,1]
head(a1)
a<-a1

#adding replicate information
replicate = as.factor(c("X04F","X04F","X04F","X04M","X04M","X04M","X13F","X13F","X13F","X13M","X13M","X13M","X13M","X15F","X15F","X15F","X15M","X15M","X15M","X21F","X21F","X21F","X21M","X21M","X21M","X21M"))

#create the DGE object including all count matrix, replicate info and annotations when available
x<-DGEList(counts=round(a),genes=rownames(a), group = replicate)
head(x)
```

# Step 3: Filter out low count
```{r}
rowSums(x$counts==0) # for each sample where the count is 0, then sums the samples that have counts = 0 for a gene
table(rowSums(x$counts==0) == 26) #number of genes that have all samples with 0 counts
proportion_0_count= 21236/2393
proportion_0_count
```
About 9% of samples have 0 counts across all samples => these can be easily filtered out.
Additionally, we need to filter out the genes that have less than 5 counts for in at least 20% of samples

To do: check filtering options in filterByExpr
```{r}
#They cluster as expected (sex -> B status -> line). We see a clear clustering by sex in the PCA of estimated counts running sleuth, but no clear clustering by B status (note that the default threshold to filter out transcripts in sleuth has been lowered to at least 5 counts in 20% of samples)

dim(x)



#compare filtering options
keep.exprs.group <- filterByExpr(x, group=x$samples$group,min.count=5, min.prop = 20) 
#default: By default, the function keeps genes with about 10 read counts or more in a minimum number of samples, where the number of samples is chosen according to the minimum group sample size. The actual filtering uses CPM values rather than counts in order to avoid giving preference to samples with large library sizes

keep.exprs.sample <- filterByExpr(x, group=rownames(x$samples),min.count=5, min.prop = 20) 

x1 <- x[keep.exprs.group, keep.lib.sizes=FALSE]
x2 <- x[keep.exprs.sample, keep.lib.sizes=FALSE]

dim(x)
dim(x1)
dim(x2) 

#using filtering by sample
x<-x2

```

## Step 4: Normalize distribution

####  MD plot for individual samples pre normalization
```{r}
pdf("rsem_gene_mdplot-prenorm.pdf", width=10, height=10, pointsize=12)
par(mfrow = c(1, 1)) # 2 rows, 2 columns
for (i in 1:ncol(x)) {
	plotMD(cpm(x, log=TRUE), column=i, xlab = "Average log-expression", ylab = "Expression log-ratio (this sample vs others)", main = colnames(x)[i])
	abline(h=0, col="red", lty=2, lwd=2)
}
dev.off()

```


####  4: Normalize distribution
```{r}
# Apply  TMM normalization
x <- calcNormFactors(x, method = "TMM")
x$samples$norm.factors
```


####  MD plot for individual samples post normalization
```{r}
pdf("rsem_gene_mdplot-postnorm.pdf", width=10, height=10, pointsize=12)
par(mfrow = c(1, 1)) # 2 rows, 2 columns
for (i in 1:ncol(x)) {
	plotMD(cpm(x, log=TRUE), column=i, xlab = "Average log-expression", ylab = "Expression log-ratio (this sample vs others)", main = colnames(x)[i])
	abline(h=0, col="red", lty=2, lwd=2)
}

dev.off()

```


#### Quality control
More plots to check distribution
```{r}
pdf("rsem_gene_qualitycontrol.pdf", width=10, height=5, pointsize=12)
par(mfrow = c(1, 2)) 

x$samples$lib.size
barplot(x$samples$lib.size,names=colnames(x),las=2)
title("Barplot of library sizes")

logcounts <- cpm(x,log=TRUE)
boxplot(logcounts, xlab="", ylab="Log2 counts per million",las=2)
abline(h=median(logcounts),col="blue")
title("Boxplots of logCPMs (normalized)")
dev.off()
```


## Step 4: MDS plot to check for grouping
```{r}
# checking grouping by sex, and by B presence

pdf("rsem_gene_genomebased_MDSplot-per-factor.pdf", width=10, height=10, pointsize=12)
par(mfrow=c(2,2))

sampleinfo2 <- sampleinfo %>% mutate(Sex = as.factor(Sex))
levels(sampleinfo2$Sex)
col.sex <- c("red","blue")[sampleinfo2$Sex]
data.frame(sampleinfo2$Sex,col.sex)
#Plot by sex
plotMDS(x,col=col.sex, main="Sex")
legend("top",fill=c("red","blue"),legend=levels(sampleinfo2$Sex))


# plot by B presence
sampleinfo2 <- sampleinfo %>% mutate(Bpresence = as.factor(Bpresence))

levels(sampleinfo2$Bpresence)
col.B <- c("grey","red")[sampleinfo2$Bpresence]
data.frame(sampleinfo2$Bpresence,col.B)
#Plot by sex
plotMDS(x,col=col.B, main="B presence")
legend("top",fill=c("grey","red"),legend=levels(sampleinfo2$Bpresence))


#plot by genotype
sampleinfo2 <- sampleinfo %>% mutate(Geno = as.factor(Geno))

levels(sampleinfo2$Geno)
col.geno <- c("green","red","blue","black")[sampleinfo2$Geno]
data.frame(sampleinfo2$Geno,col.geno)
#Plot by sex
plotMDS(x,col=col.geno, main="Genotype")
legend("top",fill=c("green","red","blue","black"),legend=levels(sampleinfo2$Geno))
dev.off()
```


Based on the groupings, male and female expression profiles are definitely different regardless of genotype of B presence. Additionally, if we group by B and no B, we see separation of groups by sex and B presence. Finally, if we group by genotype, 15 and 21 are together, which seems logical since they come from the same original population RBGE25. Female samples of 04 and 13 are together but separated when we look at males. This could indicate that in 04 and 13, we have different numbers of B and that this difference in B number only affects expression profile of males. 



## Step 5: Hierarchical clustering
```{r}
var_genes <- apply(logcounts, 1, var)
head(var_genes)
select_var <- names(sort(var_genes, decreasing=TRUE))[1:1000]
head(select_var)
highly_variable_lcpm <- logcounts[select_var,]
dim(highly_variable_lcpm)
head(highly_variable_lcpm)


## Get some nicer colours
library(RColorBrewer)
mypalette <- brewer.pal(11,"RdYlBu")

#mapping by Sex
morecols <- colorRampPalette(mypalette)

pdf("rsem_gene_genomebased_heatmapbysex.pdf", width=10, height=10, pointsize=12)

# Plot the heatmap by sex
heatmap.2(highly_variable_lcpm,col=rev(morecols(50)),trace="none", main="Top 1000 most variable genes across samples",ColSideColors=col.sex,scale="row")

dev.off()
  
# Plot the heatmap by B presence
  
pdf("rsem_gene_genomebased_heatmapbyB.pdf", width=10, height=10, pointsize=12)
heatmap.2(highly_variable_lcpm,col=rev(morecols(50)),trace="none", main="Top 1000 most variable genes across samples",ColSideColors=col.B,scale="row")
dev.off()
```




## Step 6: Model design

The model design and fit is based on sex by B presence groups. Model will be fit based on this group, then we can compare pairs of groups by creating different contrast matrices.

### Define groups by sex and B presence or absence
```{r}
group1=c("FB","FB","FB","MB","MB","MB","FB","FB","FB","MB","MB","MB","MB","FnoB","FnoB","FnoB","MnoB","MnoB","MnoB","FnoB","FnoB","FnoB","MnoB","MnoB","MnoB","MnoB")
```

### Design: Model Matrix by group defined above
```{r}
design1 <- model.matrix(~0 + group1)
colnames(design1)
rownames(design1) = rownames(x$samples)
```


### Removing heteroscedascity from count data
Voom plot

```{r}
dim(x)
par(mfrow=c(1,1))
v1 <- voom(x,design1,plot = TRUE)
v1
```


### limma lm fit
```{r}
fit1 <- lmFit(v1)
```

### Comparison 1: all transcripts that are just B male
#### Contrast matrix: called fit.cont1
```{r}
colnames(design1)
# I only want transcripts differentially expressed in male B samples.
cont.matrix1 <- makeContrasts(BmalevnoBmale = group1MB - group1MnoB, BmalevsfemaleB = group1MB - group1FB, BmalevsfemalenoB = group1MB - group1FnoB,BfemalevsnoBfemale = group1FB - group1FnoB, noBmalevsnoBfemale = group1MnoB - group1FnoB, levels=design1)

cont.matrix1



fit.cont1 <- contrasts.fit(fit1, cont.matrix1)
fit.cont1 <- eBayes(fit.cont1)

```



#### Compare mean variance trend

```{r}
pdf("rsem_gene_genomebased_meanvariancetrends.pdf", width=10, height=5, pointsize=12)
par(mfrow=c(1,2))
v1 <- voom(x,design1,plot = TRUE)
plotSA(fit.cont1, main="Final model: Mean-variance trend")
dev.off()
```


#### Examine the number of DE genes
```{r}
tfit <- treat(fit.cont1, lfc=1) 

dt <- decideTests(tfit)
summary(dt)

#extracting only the Bmale specific in de.common
de.common <- which(dt[,1]!=0 & dt[,2]!=0 &dt[,3]!=0)
length(de.common) # B transcripts

pdf("rsem_gene_genomebased_vennDEcommon.pdf", width=10, height=10, pointsize=12)
par(mfrow=c(2,2))

#Venn Diagram for B vs no B in males
vennDiagram(dt[,1], circle.col=c("turquoise", "salmon","orange"),include=c("up","down"))

#Venn Diagram for B vs no B in females
vennDiagram(dt[,4], circle.col=c("turquoise", "salmon","orange"),include=c("up","down"))

#Venn Diagram for Bmale
vennDiagram(dt[,1:3], circle.col=c("turquoise", "salmon","orange"),include=c("up","down"))
dev.off()

#list of genes from Bmale
fit.cont1$genes[de.common,]


```
de.common contains the genes that overexpressed only in B males.

P values for all comparisons from the contrast matrix
```{r}

#retrieving p val for all DE genes
maleB.noB<-topTable(fit.cont1, coef=1,number = summary(dt)[1]+summary(dt)[3])
maleB.femaleB<-topTable(fit.cont1, coef=2, number = summary(dt)[4]+summary(dt)[6]) 
maleB.femalenoB<-topTable(fit.cont1, coef=3, number = summary(dt)[7]+summary(dt)[9] ) 
femaleB.femalenoB<-topTable(fit.cont1, coef=4, number = summary(dt)[10]+summary(dt)[12]) 
malenoB.femalenoB<-topTable(fit.cont1, coef=5, summary(dt)[13]+summary(dt)[15]) 

#add annotation for each list and save to csv
maleB.noB_df<-data.frame(maleB.noB)
maleB.noB_annot = merge(maleB.noB,annot, by.x='genes', by.y='gid',all.x=TRUE)
write.csv(maleB.noB_annot,"rsem_gene_maleB.noB_annot.csv")

maleB.femaleB_df<-data.frame(maleB.femaleB)
maleB.femaleB_annot = merge(maleB.femaleB,annot, by.x='genes', by.y='gid',all.x=TRUE)
write.csv(maleB.femaleB_annot,"rsem_gene_maleB.femaleB.csv")

maleB.femalenoB_df<-data.frame(maleB.femalenoB)
maleB.femalenoB_annot = merge(maleB.femalenoB,annot, by.x='genes', by.y='gid',all.x=TRUE)
write.csv(maleB.femalenoB_annot,"rsem_gene_maleB.femalenoB.csv")

femaleB.femalenoB_df<-data.frame(femaleB.femalenoB)
femaleB.femalenoB_annot = merge(femaleB.femalenoB,annot, by.x='genes', by.y='gid',all.x=TRUE)
write.csv(femaleB.femalenoB_annot,"rsem_gene_femaleB.femalenoB.csv")

malenoB.femalenoB_df<-data.frame(malenoB.femalenoB)
malenoB.femalenoB_annot = merge(malenoB.femalenoB,annot, by.x='genes', by.y='gid',all.x=TRUE)
write.csv(malenoB.femalenoB_annot,"rsem_gene_malenoB.femalenoB.csv")

```

#### add gene annotation to de.common
```{r}
Bmale = data.frame(fit.cont1$genes[de.common,])
colnames(Bmale) =  "gene"
Bmale_annot = merge(Bmale,annot, by.x='gene', by.y='gid',all.x=TRUE)

Bmale_annot2<-Bmale_annot[!is.na(Bmale_annot$V2), ]
write.csv(Bmale_annot2,"rsem_gene_Bmale_annot2.csv")
```

#### MDplots for fit.cont1
```{r}
pdf("rsem_gene_genomebased_MDplots.pdf", width=20, height=10, pointsize=12)
par(mfrow=c(2,3))
plotMD(fit.cont1, column=1, status=dt[,1], main=colnames(fit.cont1)[1], 
       xlim=c(-8,13),cex=0.5)

plotMD(fit.cont1, column=2, status=dt[,2], main=colnames(fit.cont1)[2], 
       xlim=c(-8,13),cex=0.5)

plotMD(fit.cont1, column=3, status=dt[,3], main=colnames(fit.cont1)[3], 
       xlim=c(-8,13),cex=0.5)

plotMD(fit.cont1, column=3, status=dt[,4], main=colnames(fit.cont1)[4], 
       xlim=c(-8,13),cex=0.5)

plotMD(fit.cont1, column=3, status=dt[,5], main=colnames(fit.cont1)[5], 
       xlim=c(-8,13),cex=0.5)
dev.off()
```

